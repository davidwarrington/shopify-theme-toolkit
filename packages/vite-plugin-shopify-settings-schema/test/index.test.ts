import fs from 'node:fs/promises';
import path from 'node:path';
import { build } from 'vite';
import { describe, expect, it } from 'vitest';
import shopifySettingsSchema from '../src';

function getFixture(fixture: string) {
  const basePath = path.join(__dirname, '__fixtures__', fixture);
  const input = path.join(basePath, 'config', 'settings_schema.ts');
  const output = path.join(basePath, 'config', 'settings_schema.json');

  return {
    build() {
      return build({
        root: basePath,
        logLevel: 'silent',
        build: {
          emptyOutDir: false,
          outDir: basePath,
          rollupOptions: {
            input: [input],
          },
        },
        plugins: [
          shopifySettingsSchema({
            input,
            output,
          }),
        ],
      });
    },
    async getResult<T>() {
      const content = await fs.readFile(output, { encoding: 'utf-8' });
      return JSON.parse(content) as T;
    },
  };
}

describe('vite-plugin-shopify-settings-schema', () => {
  it('can output a config with a single module', async () => {
    const fixture = getFixture('with-single-module');
    await fixture.build();

    const result = await fixture.getResult();

    expect(result).toMatchSnapshot();
  });

  it('can output a config with modules', async () => {
    const fixture = getFixture('with-modules');
    await fixture.build();

    const result = await fixture.getResult();

    expect(result).toMatchSnapshot();
  });
});
