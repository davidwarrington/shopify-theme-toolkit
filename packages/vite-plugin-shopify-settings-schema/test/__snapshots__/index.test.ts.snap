// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`vite-plugin-shopify-settings-schema > can output a config with a single module 1`] = `
{
  "name": "vite-plugin-shopify-settings-schema",
  "tests": [
    "with-single-module",
  ],
}
`;

exports[`vite-plugin-shopify-settings-schema > can output a config with modules 1`] = `
{
  "name": "vite-plugin-shopify-settings-schema",
  "tests": [
    "with-modules",
  ],
}
`;

exports[`vite-plugin-shopify-settings-schema > can output a config with other filetypes 1`] = `
{
  "name": "vite-plugin-shopify-settings-schema",
  "tests": [
    "with-other-filetypes",
  ],
}
`;

exports[`vite-plugin-shopify-settings-schema > can output a config with src and dist dirs 1`] = `
{
  "name": "vite-plugin-shopify-settings-schema",
  "tests": [
    "with-src-dist-dirs",
  ],
}
`;
